Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -52,16 +57,24 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
 
     let context = initialData.map(\.context)
 
-    self.configurePaymentMethodsViewControllerWithValue = initialData
+    let configurePaymentMethodsData = Signal.merge(
+      initialData,
+      initialData.takeWhen(self.userSessionStartedSignal)
+    )
+
+    self.configurePaymentMethodsViewControllerWithValue = configurePaymentMethodsData
       .compactMap { data -> PledgePaymentMethodsValue? in
         guard let user = AppEnvironment.current.currentUser else { return nil }
         guard let reward = data.rewards.first else { return nil }
 
         return (user, data.project, reward, data.context, data.refTag)
       }
 
-    // TODO: Respond to login flow.
-    let isLoggedIn = initialData.ignoreValues()
+    self.goToLoginSignup = Signal.combineLatest(initialData, self.goToLoginSignupSignal)
```

Comment:
Is `Signal.combineLatest(a, b).map(first)` effectively just a `takeWhen`? If so, I find `takeWhen` clearer.

Reply:
I don't think so, because we want to update whenever either a or b has a new event. We could do a `Signal.merge(a, a.takeWhen(b))` instead, though I don't really think that's cleaner. I'm okay with switching to that if you like it better? I was just following our usual pattern for the `configureWithData` signal here (and copy-pasting from the normal pledge VC). Or if you have any better ideas for a thing that would make this doable in one command, let me know!

Reply:
Nah, you're right, it's not any clearer, then. I guess it would be nice to have an operator like `take(value: someSignal, whenAnyChange: [someSignal, anotherSignal, aThirdSignal])` but that is vastly out of scope here.

Reply:
You were right about this one, actually! I just mixed up use-cases and made us both confused 😅